bebraven/platform
# Fix the LRS proxy to handle the activities/state endpoint.

Articulate Rise360 xAPI packages call GET and PUT to the xAPI
state API endpoint with three types of state's to save and store:
bookmark, cumulative_time, and suspend_data.

Both the GET and PUT use the 'Content-Type: application/octet-stream'
header when sending or receiving the payload, even though the suspend_data
one is actually valid JSON.

This commit specifically looks for calls hitting that end-point and sets the
payload and header in the request/response appropriately.

Note: the NaN hack/fix in there no longer appears to be valid now. I put a log statement in there
and did a lesson but it was never hit so I removed that code and those specs.

Also note: using `raw_post` instead of `request_parameters` seems to be cleaner
and more what we want. `request_parameters` is info about the request after Rack
has already processed it which is why there is that weird stuff we have to hack around.
`raw_post` is just the raw body of the request. I removed all the specs related to that hackery as well.

### Test Plan:
- Make sure Lessons pick up where they left off when you leave and
  come back. Note that it's not exact, it picks up where you left off
  to the nearest bookmarked item in the sidebar. If you're on a Mastery Quiz
  part of the Lesson, it should remember where in the quiz you are and what
  partial answer's you've given on the current quiz question.
- Make sure Projects still send and receive xAPI statemenst to the LRS.
- `bundle exec spec/lib/lrs_xapi_proxy_spec.rb`
